src/components/studies/participants/ParticipantManager.tsx

```diff
@@ -206,34 +208,111 @@ const ParticipantManager: FunctionComponent<ParticipantManagerProps> = () => {
 
   const {data: scheduleEvents = [], error: eventError} = useEvents(studyId)
 
-  const {
-    data,
-    status,
-    error,
-    run,
-    setData: setParticipantData,
-  } = useAsync<ParticipantData>({
-    status: 'PENDING',
-    data: null,
-  })
-
-  React.useEffect(() => {
-    if (!study?.identifier) {
-      return
-    }
-    const fn = async () => {
-      const result: ParticipantData = await run(
-        ParticipantUtility.getParticipants(
-          study.identifier,
+  // const {
+  //   data,
+  //   status,
+  //   error,
+  //   run,
+  //   setData: setParticipantData,
+  // } = useAsync<ParticipantData>({
+  //   status: 'PENDING',
+  //   data: null,
+  // })
+  // console.log("DATA",data?.items)
+
+  const {data: data, error : participantError, status: status} = useParticipants(study?.identifier,currentPage,pageSize,tab)
+  
+  const {    
+    mutate,
+    isSuccess,
+    isError,
+    isLoading: isParticipantUpdating,
+  } = useUpdateParticipantInList()
+  
+  const [curStatus, setCurStatus] = React.useState<RequestStatus>('PENDING')
+  // console.log("data", curStatus)
+
+  React.useEffect(()=>{
+    if(isParticipantUpdating || status ==='loading') setCurStatus('PENDING')
+    else if(isError) setCurStatus('REJECTED')
+    else setCurStatus('RESOLVED')
+  },[isParticipantUpdating, status])
+  
+  const onAction = async (
+    studyId:string,
+    type:'WITHDRAW' | 'DELETE' | 'UPDATE', 
+    currentPage?: number, 
+    pageSize?: number, 
+    tab?: ParticipantActivityType,
+    userId?:string,  
+    note?: string, 
+    updatedFields?: any,
+    customEvents?: ParticipantEvent[]
+    ) => {
+
+    switch(type){
+      case 'WITHDRAW':
+        await mutate({
+          action: type, 
+          studyId, 
+          userId, 
+          note,
           currentPage,
           pageSize,
-          tab,
-          token!
-        )
-      )
+          tab})
+        setRefreshParticipantsToggle(prev => !prev)
+        return
+      case 'DELETE':
+        setLoadingIndicators(_ => ({isDeleting: true}))
+
+        for(let i = 0; i < selectedParticipantIds[tab!].length; i++){
+          await mutate({
+            action: type, 
+            studyId,
+            userId: selectedParticipantIds[tab!][i],
+            currentPage,
+            pageSize,
+            tab})
+        }
+        setLoadingIndicators(_ => ({isDeleting: false}))
+        setDialogState({dialogOpenRemove: false, dialogOpenSMS: false})
+        setRefreshParticipantsToggle(prev => !prev)
+        return
+      case 'UPDATE':
+        await mutate({
```

Comment:
I don't think those awaits will work. You can console.log to confirm. You can specify success/error callbacks (preferable) or use mutateAsync (if that does not work)